Suricata: improve setup/config for ETPRO ruleset

adepasquale opened the pull request:
### PROBLEM (see #487)
- ET rule files start with the `emerging-*` prefix;
- ETPRO rule files do not start with that prefix.

### SOLUTION
Concatenate all rule files, with the exception of `*disabled.rules`, into a single rule file at `/etc/suricata/rules/tpotce.rules`

Another thing worth considering is to use "suricata-update" instead:
https://suricata-update.readthedocs.io/en/latest/quickstart.html

adepasquale commented on docker/suricata/dist/update.sh:
```diff
@@ -27,6 +27,22 @@ if [ "$myOINKCODE" != "" ] && [ "$myOINKCODE" == "OPEN" ];
+  # - removing unnecessary empty/comment lines.
+  ls /tmp/rules/*.rules | grep -v deleted.rules | while read f;
+    do
+      cat $f | sed "s/^#alert/alert/" | grep -Ev "^(#|$)" >> /tmp/tpotce.rules
```
I have left as-is the `sed` command that enables all commented-out rules (usually the performance-heavy ones) since you mentioned
> never had any issues with enabling all the rules

Another option would be to turn this into a configuration variable like `OINKCODE`

t3chn0m4g3 commented:
@adepasquale Thanks for taking the time. If you believe `suricata-update` is the best option, then we should use that instead, just let me know.

adepasquale commented:
@t3chn0m4g3 I mentioned `suricata-update` since it's already installed in your container, and downloads ET open by default:
```
$ docker-compose exec suricata /usr/bin/suricata-update
25/11/2020 -- 15:24:14 - <Info> -- Using data-directory /var/lib/suricata.
25/11/2020 -- 15:24:14 - <Info> -- Using Suricata configuration /etc/suricata/suricata.yaml
25/11/2020 -- 15:24:14 - <Info> -- Using /usr/share/suricata/rules for Suricata provided rules.
25/11/2020 -- 15:24:14 - <Info> -- Found Suricata version 5.0.3 at /usr/bin/suricata.
25/11/2020 -- 15:24:14 - <Info> -- Loading /etc/suricata/suricata.yaml
25/11/2020 -- 15:24:14 - <Info> -- Disabling rules for protocol enip
25/11/2020 -- 15:24:14 - <Info> -- No sources configured, will use Emerging Threats Open
25/11/2020 -- 15:24:14 - <Info> -- Checking https://rules.emergingthreats.net/open/suricata-5.0.3/emerging.rules.tar.gz.md5.
25/11/2020 -- 15:24:15 - <Info> -- Remote checksum has not changed. Not fetching.
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/app-layer-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/decoder-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/dhcp-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/dnp3-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/dns-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/files.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/http-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/ipsec-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/kerberos-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/modbus-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/nfs-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/ntp-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/smb-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/smtp-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/stream-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Loading distribution rule file /usr/share/suricata/rules/tls-events.rules
25/11/2020 -- 15:24:15 - <Info> -- Ignoring file rules/emerging-deleted.rules
25/11/2020 -- 15:24:18 - <Info> -- Loaded 28534 rules.
25/11/2020 -- 15:24:18 - <Info> -- Disabled 0 rules.
25/11/2020 -- 15:24:18 - <Info> -- Enabled 0 rules.
25/11/2020 -- 15:24:18 - <Info> -- Modified 0 rules.
25/11/2020 -- 15:24:18 - <Info> -- Dropped 0 rules.
25/11/2020 -- 15:24:18 - <Info> -- Enabled 145 rules for flowbit dependencies.
25/11/2020 -- 15:24:18 - <Info> -- Backing up current rules.
25/11/2020 -- 15:24:21 - <Info> -- Writing rules to /var/lib/suricata/rules/suricata.rules: total: 28534; enabled: 21285; added: 0; removed 0; modified: 14
25/11/2020 -- 15:24:21 - <Info> -- Testing with suricata -T.
25/11/2020 -- 15:24:34 - <Info> -- Done.

t3chn0m4g3 commented:
@adepasquale Yes, at some point it was installed by default even when compiling from source, does not enable all the rules, though which needs to be done still manually.

adepasquale commented:
Please disregard in favor of #727
